src/student_success_tool/analysis/pdp/targets/shared.py

```diff
@@ -280,6 +280,74 @@ def _compute_intensity_num_terms(
         for intensity, time, unit in intensity_time_lefts
     ]
 
+def get_nth_student_terms(
+    df: pd.DataFrame,
+    *,
+    student_id_cols: str | list[str] = "student_guid",
+    sort_cols: str | list[str] = "term_rank",
+    n: int,
+    include_cols: t.Optional[list[str]] = None,
+) -> pd.DataFrame:
+    """
+    For each student, get the nth row in ``df`` (in ascending order of ``sort_cols`` )
+    and a configurable subset of columns. 
+
+    Args:
+        df
+        student_id_cols
+        sort_cols
+        n
```

Comment:
You should document what `n` is here -- notably, that `n=0` => get the first term, i.e. equivalent to the functionality of `get_first_student_terms()`.

Reply:
@bdewilde do you want to scrap the get_first_student_terms function and just utilize this with a default n=0 instead since it's universal / more flexible?

Reply:
We could consolidate functionality under the hood at some point, but not here in this PR, and it's probably not necessary anyway -- pandas has both `.first()` and `.nth()` for a reason :)